silo/tests/test_views.py

```diff
@@ -138,3 +147,131 @@ def test_index_get_unauthenticated_no_activity_url(self):
         request.user = AnonymousUser()
         with self.assertRaises(ImproperlyConfigured):
             views.IndexView.as_view()(request)
+
+
+class ExportViewsTest(TestCase, MongoTestCase):
+    def setUp(self):
+        factories.ReadType(read_type='CustomForm')
+        self.tola_user = factories.TolaUser()
+        self.factory = APIRequestFactory()
+        self.silo_id = None
+
+    def tearDown(self):
+        if self.silo_id:
+            # Have to remove the created lvs
+            lvss = LabelValueStore.objects.filter(silo_id=self.silo_id)
+            for lvs in lvss:
+                lvs.delete()
+
+    def test_export_csv_superuser(self):
+        self.tola_user.user.is_staff = True
+        self.tola_user.user.is_superuser = True
+        self.tola_user.user.save()
+
+        # Create the Silo to store the data
+        wflvl1 = factories.WorkflowLevel1(
+            organization=self.tola_user.organization)
+        fields = [
+            {
+                'name': 'color',
+                'type': 'text'
+            },
+            {
+                'name': 'type',
+                'type': 'text'
+            }
+        ]
+        meta = {
+            'name': 'Export Test',
+            'description': 'This is a test.',
+            'fields': json.dumps(fields),
+            'level1_uuid': wflvl1.level1_uuid,
+            'tola_user_uuid': self.tola_user.tola_user_uuid
+        }
+        request = self.factory.post('', data=meta)
+        request.user = self.tola_user.user
+        view = CustomFormViewSet.as_view({'post': 'create'})
+        response = view(request)
+        self.assertEqual(response.status_code, 201)
+        # For the tearDown
+        self.silo_id = response.data['id']
+        silo = Silo.objects.get(id=self.silo_id)
+        read = silo.reads.all()[0]
+
+        # Upload data
+        data = [{
+            'color': 'black',
+            'type': 'primary'
+        }, {
+            'color': 'white',
+            'type': 'primary'
+        }, {
+            'color': 'red',
+            'type': 'primary'
+        }]
+        util.saveDataToSilo(silo, data, read)
+
+        # Export to CSV
+        request = self.factory.get('')
+        request.user = self.tola_user.user
+        response = views.export_silo(request, self.silo_id)
+        self.assertEqual(response.status_code, 200)
+        self.assertIn('color,type', response.content)
+        self.assertIn('black,primary', response.content)
+
+    def test_export_csv_normaluser(self):
+        # The user has to be a superuser to create a custom form
+        user = factories.User(first_name='John', last_name='Lennon',
+                              is_staff=True, is_superuser=True)
```

Comment:
This is wrong for a normal user

Reply:
Double check the users of the requests. It's not wrong.

Reply:
As I said, to create a custom form you need to be a superuser.